backends/tfhe-cuda-backend/cuda/src/pbs/bootstrapping_key.cuh

```diff
@@ -251,4 +285,92 @@ void cuda_convert_lwe_programmable_bootstrap_key(cudaStream_t stream,
   cudaFreeHost(h_bsk);
 }
 
+template <class params>
+void convert_and_transform_128(cudaStream_t stream, uint32_t gpu_index,
+                               double *d_bsk, __uint128_t const *d_standard,
+                               uint32_t number_of_samples) {
+
+  size_t required_shared_memory_size = sizeof(double) * params::degree / 2 * 4;
+  int grid_size = number_of_samples;
+  int block_size = params::degree / params::opt;
+  bool full_sm =
+      (required_shared_memory_size <= cuda_get_max_shared_memory(gpu_index));
+  size_t buffer_size =
+      full_sm ? 0 : (size_t)number_of_samples * params::degree / 2 * 4;
+  size_t shared_memory_size = full_sm ? required_shared_memory_size : 0;
+  double *buffer = (double *)cuda_malloc_async(buffer_size, stream, gpu_index);
+
+  // configure shared memory for batch fft kernel
+  if (full_sm) {
+    check_cuda_error(cudaFuncSetAttribute(
+        batch_NSMFFT_strided_128<FFTDegree<params, ForwardFFT>, FULLSM>,
+        cudaFuncAttributeMaxDynamicSharedMemorySize, shared_memory_size));
+    check_cuda_error(cudaFuncSetCacheConfig(
+        batch_NSMFFT_strided_128<FFTDegree<params, ForwardFFT>, FULLSM>,
+        cudaFuncCachePreferShared));
+  }
+
+  // convert u128 into 4 x double
+  batch_convert_u128_to_f128_strided_as_torus<params>
```

Comment:
I'm not sure it's necessary to keep strided in the names, as anyway this FFT implementation can only work with the data layout that we use? Wdyt?

Reply:
we have two options  for it strided and non strided. lets discuss it.